From dc1262b3ccb7302b040a56f2aa42a4b5869b00a8 Mon Sep 17 00:00:00 2001
From: Serge Hallyn <serge.hallyn@ubuntu.com>
Date: Tue, 8 Mar 2016 13:44:38 -0800
Subject: tests: set clone_children if need be

Lxc only sets it on /lxc, not on /.

It's conceivable that we should really re-set this to the original
value, to prevent making later tests not fail when they should.  I
didn't do that.

Signed-off-by: Serge Hallyn <serge.hallyn@ubuntu.com>
---
 src/tests/lxc-test-apparmor-mount | 1 +
 src/tests/lxc-test-unpriv         | 1 +
 src/tests/lxc-test-usernic.in     | 1 +
 3 files changed, 3 insertions(+)

diff --git a/src/tests/lxc-test-apparmor-mount b/src/tests/lxc-test-apparmor-mount
index 0c5c7cd..3fd3f5a 100755
--- a/src/tests/lxc-test-apparmor-mount
+++ b/src/tests/lxc-test-apparmor-mount
@@ -131,6 +131,7 @@ elif [ -e /sys/fs/cgroup/cgmanager/sock ]; then
 	done
 else
 	for d in /sys/fs/cgroup/*; do
+		[ -f $d/cgroup.clone_children ] && echo 1 > $d/cgroup.clone_children
 		[ ! -d $d/lxctest ] && mkdir $d/lxctest
 		chown -R $TUSER: $d/lxctest
 		echo $$ > $d/lxctest/tasks
diff --git a/src/tests/lxc-test-unpriv b/src/tests/lxc-test-unpriv
index bb3660e..f3a7910 100755
--- a/src/tests/lxc-test-unpriv
+++ b/src/tests/lxc-test-unpriv
@@ -112,6 +112,7 @@ elif [ -e /sys/fs/cgroup/cgmanager/sock ]; then
 	done
 else
 	for d in /sys/fs/cgroup/*; do
+		[ -f $d/cgroup.clone_children ] && echo 1 > $d/cgroup.clone_children
 		[ ! -d $d/lxctest ] && mkdir $d/lxctest
 		chown -R $TUSER: $d/lxctest
 		echo $$ > $d/lxctest/tasks
diff --git a/src/tests/lxc-test-usernic.in b/src/tests/lxc-test-usernic.in
index a38beb9..6a5650d 100755
--- a/src/tests/lxc-test-usernic.in
+++ b/src/tests/lxc-test-usernic.in
@@ -104,6 +104,7 @@ elif [ -e /sys/fs/cgroup/cgmanager/sock ]; then
 	done
 else
 	for d in /sys/fs/cgroup/*; do
+		[ -f $d/cgroup.clone_children ] && echo 1 > $d/cgroup.clone_children
 		[ ! -d $d/lxctest ] && mkdir $d/lxctest
 		chown -R usernic-user: $d/lxctest
 		echo $$ > $d/lxctest/tasks
